crates/compiler/mono/src/tail_recursion.rs

```diff
@@ -907,13 +907,13 @@ impl<'a> TrmcEnv<'a> {
                                 reuse: None,
                             };
 
-                            let let_tag = |next| Stmt::Let(*symbol, tag_expr, *layout, next);
+                            let indices = vec![in env.arena; cons_info.tag_id as u64, recursive_field_index as u64].into_bump_slice();
```

Comment:
using `arena.alloc([cons_info.tag_id as u64, recursive_field_index as u64])` is slightly more efficient here